src/drivers/cmakeDriver.ts

```diff
@@ -305,6 +305,21 @@ export abstract class CMakeDriver implements vscode.Disposable {
         }
     }
 
+    /**
+     * Get the environment variables that should be set at CPack and packaging time.
+     */
+    async getCPackCommandEnvironment(): Promise<Environment> {
+        if (this.useCMakePresets) {
+            return EnvironmentUtils.create(this._packagePreset?.environment);
+        } else {
+            let envs = this._kitEnvironmentVariables;
+            envs = EnvironmentUtils.merge([envs, await this.computeExpandedEnvironment(this.config.environment, envs)]);
+            envs = EnvironmentUtils.merge([envs, await this.computeExpandedEnvironment(this.config.testEnvironment, envs)]);
```

Comment:
This shouldn't be here, this is the `testEnvironment`, but this is the method for the cpack. 

Additionally, in the `useCMakePresets` branch, we now support using the settings to adjust the environment. This should be added.

Reply:
Thank you for reminding me of the more recent work done regarding settings and overrides. For the first IF path, I applied what I found is new in the CMake Tools code base. For the ELSE path, indeed I forgot code from the CTest case, which also has non-preset functionality but CPack does not function in that case.